interview-questions.md

```diff
@@ -7,63 +7,138 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+
   Researched answer:
 
+  paired programming is also called "programming out loud." both the driver and navigator must be actively participating or else no benefit can be expected.
+  both are expected to keep an active commentary.
+  the navigator is focused on overall direction and the driver is typing
+  https://www.agilealliance.org/glossary/pairing/
+
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: Test Driven Development is a part of agile development. its the process of developing a product, and you go back and forth between coding and testing. If you need to add a new feature or something and you add it to the code then you can use TDD to test it as you go and make sure its not creating any bugs
 
   Researched answer:
 
+  there are 2 simple steps to TDD
+  1. introduce a test failure
+  2. write minimal code required to pass the testing
+
+  there are a few rules that go along with TDD as well.
+
+  1. You are not allowed to write any production code unless it is to make a failing unit test pass.
+
+  2. You are not allowed to write any more of a unit test than is sufficient to fail; and compilation failures are failures.
+
+  3. You are not allowed to write any more production code than is sufficient to pass the one failing unit test.
+
+  https://www.codementor.io/@copperstarconsulting/test-driven-workflow-in-two-easy-steps-bfjln9sl4
 
 
 7. What is something we did in class this week you found helpful?  
 
   Your answer:
 
+  when we did that review of git/github! it was really helpful to actually apply it in context. It was way too confusing on the first day because I felt like there was no context
+
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
 - Classes
+
+    Classes are templates for Javascript objects
+    they are not objects- just templates. You can use a class to create objects
+    the keyword 'class' creates a class.
+    then, you add a constructor method. if you don't add one then javascript adds an empty constructor method
+
+
 - React
+
+  React is a JavaScript library for building user interfaces.
+
+  React is used to build single page applications.
+
+  React allows us to create reusable UI components.
+
+  theres a terminal command called create-react-app that allows you to create apps in react
+
+  react's goal is to render HTML in a web page
+
+  you can create components in react, and they are independent/reusable bits of code that serve the same purpose as functions in Javascript. props are arguments passed into react components via HTML attributes
+
 - React State
-- CRUD
+
+  React components have a built in state object. the State object is where you store property values that belong to the component. when a state object changes, the component re-renders
+
+
+- CRUD -
+    4 methods of react.
+    the methods respectively stand for Create, Render, Update, delete
```

Comment:
really excellent work on this section! your answers are well-written and your research is super solid.